include/ensmallen_bits/moead/moead_impl.hpp

```diff
@@ -0,0 +1,369 @@
+  }
+
+  return performance;
+}
+
+//! Perform mutation of the candidate.
+template<typename MatType>
+inline void MOEAD::Mutate(MatType& child,
+    const double& rate,
+    const arma::vec& lowerBound,
+    const arma::vec& upperBound)
+{
+  size_t numVariables = lowerBound.n_elem;
+  double rnd, delta1, delta2, mutationPower, deltaq;
+  double current, currentLower, currentUpper, value, upperDelta;
+
+  for (size_t j = 0; j < numVariables; ++j)
+  {
+    double determiner = arma::randu();
+    if (determiner <= rate)
+    {
+      current = child[j];
+      currentLower = lowerBound(j);
+      currentUpper = upperBound(j);
+      delta1 = (current - currentLower) / (currentUpper - currentLower);
+      delta2 = (currentUpper - current) / (currentUpper - currentLower);    
+      rnd = arma::randu();
+      mutationPower= 1 / distributionIndex;
+      if (rnd <= 0.5)
+      {
+        upperDelta = 1.0 - delta1;
+        value = 2.0 * rnd + (1.0 - 2.0 * rnd) *
+            (std::pow(upperDelta, (distributionIndex + 1.0)));
+        deltaq = std::pow(value, mutationPower) - 1.0;
+      }
+      else
+      {
+        upperDelta = 1.0 - delta2;
+        value = 2.0 * (1.0 - rnd) + 2.0 * (rnd - 0.5) *
+            (pow(upperDelta, (distributionIndex + 1.0)));
+        deltaq = 1.0 - (pow(value, mutationPower));
+      }
+      current = current + deltaq * (currentUpper - currentLower);
+      if (current < currentLower) current = currentLower;
+      if (current > currentUpper) current = currentUpper;
+      child[j] = current;
+    }
+  }
+}
+
+//! Calculate the output for single objective function using the Tchebycheff
+//! approach.
+inline double MOEAD::DecomposedSingleObjective(const arma::vec& weights,
+    const arma::vec& idealPoint,
+    const arma::vec& evaluatedCandidate)
+{
+  return arma::min(weights % arma::abs(evaluatedCandidate - idealPoint));
+}
+
+inline bool MOEAD::Dominates(const arma::vec& first,
+    const arma::vec& second)
+{
+  int flag = 0;
+  for (size_t i = 0; i < numObjectives; ++i)
+  {
+    if (first(i) > second(i))
+      return false;
+
+    if(first(i) < second(i))
+      flag = 1;
+  }
+  if (flag)
+    return true;
+
+  return false;
+}
+
+//! No objectives to evaluate.
+template<std::size_t I,
+  typename MatType,
+  typename ...ArbitraryFunctionType>
+  typename std::enable_if<I == sizeof...(ArbitraryFunctionType), void>::type
+MOEAD::EvaluateObjectives(
+    std::vector<MatType>&,
+    std::tuple<ArbitraryFunctionType...>&,
+    arma::mat &)
+{
+  // Nothing to do here.
+}
+
+//! Evaluate the objectives for the entire population.
+template<std::size_t I,
+  typename MatType,
+  typename ...ArbitraryFunctionType>
+  typename std::enable_if<I < sizeof...(ArbitraryFunctionType), void>::type
+MOEAD::EvaluateObjectives(
+    std::vector<MatType>& population,
+    std::tuple<ArbitraryFunctionType...>& objectives,
+    arma::mat& calculatedObjectives)
+{
+  for (size_t i = 0; i < population.size(); ++i)  {
```

Comment:
```suggestion
  for (size_t i = 0; i < population.size(); ++i)
  {
```

The opening `{` should be on a new line.